### PR TITLE
Allow to reverse the order of colours in a brewer palette

### PR DESCRIPTION
**Pull request by jiho**
Use the same argument that hue_pal uses: direction

### CODE REVIEW

**Comment by hadley** [comment]
Could you please merge/rebase and add a bullet to NEWS?

**Comment by jiho** [comment]
Done, and squashed everything into one commit.

**Comment by jiho** [comment]
I forgot to write the documentation before. Sorry. `check()` still does not pass on my machine because of problems in creating the PDF documentation. However there is no more info and I don't really know what to fix. Hopefully that's a glitch of my machine and it'll pass on Travis.

**Comment by jiho** [comment]
It passed! It'll be great if you can merge it before the release so that I can depend on it in ggplot2.

File: R/pal-brewer.r [hunk 1]
```diff
@@ -25,9 +28,14 @@ brewer_pal <- function(type = "seq", palette = 1) {
...
-      suppressWarnings(RColorBrewer::brewer.pal(n, pal))[seq_len(n)]
+      pal <- suppressWarnings(RColorBrewer::brewer.pal(n, pal))[seq_len(n)]
     else
-      RColorBrewer::brewer.pal(n, pal)[seq_len(n)]
+      pal <- RColorBrewer::brewer.pal(n, pal)[seq_len(n)]
+
+    if (direction == -1)
+      pal <- rev(pal)
+
+    return(pal)
```
Comment (hadley) [inline R/pal-brewer.r]: For future reference, I prefer to use `return()` only for early returns

**Comment by hadley** [comment]
Thanks!

**Comment by hadley** [comment]
BTW if you have a little more time, I'd love a PR with some tests :)

File: R/pal-brewer.r (hunk 1 above)
Comment (jiho) [inline R/pal-brewer.r]: OK. I thought it was good practice to always have an explicit `return()`(it's certainly less confusing for newcomers/students in my experience).

**Comment by jiho** [comment]
What would you want me to test in this particular instance? The order of colors returned? The values of colors returned?

**Comment by jiho** [comment]
Thanks for merging it anyway, I'll get to work on the ggplot end now!

**Pull request merged**